### ISSUE TITLE
AssertionError: expected FunctionType found method

### ISSUE THREAD

**Issue by johnnynunez** [Lightning-AI/lightning#16395]
### Bug description

Conda environment 3.10 python version
Pytorch installation:
```
pip3 install --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cu118
```
Lightning
```
pip3 install https://github.com/Lightning-AI/lightning/archive/refs/heads/master.zip -U
```
PROJECT
```
https://github.com/johnnynunez/lightning-hydra-template
```
Packages
```
pip3 install -U -r requirements.txt
```

### How to reproduce the bug

```python
if cfg.get("train"):
        log.info("Starting training!")
        if cfg.get("compile"):
            log.info("Compiling model!")
            model = torch.compile(model)
        trainer.fit(model=model, datamodule=datamodule, ckpt_path=cfg.get("ckpt_path"))
```
```python
class MNISTLitModule(LightningModule):
    """Example of LightningModule for MNIST classification.

    A LightningModule organizes your PyTorch code into 6 sections:
        - Computations (init)
        - Train loop (training_step)
        - Validation loop (validation_step)
        - Test loop (test_step)
        - Prediction Loop (predict_step)
        - Optimizers and LR Schedulers (configure_optimizers)

    Docs:
        https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html
    """

    def __init__(
        self,
        net: torch.nn.Module,
        optimizer: torch.optim.Optimizer,
        scheduler: torch.optim.lr_scheduler,
    ):
        super().__init__()

        # this line allows to access init params with 'self.hparams' attribute
        # also ensures init params will be stored in ckpt
        self.save_hyperparameters(logger=False)

        self.net = net

        # loss function
        self.criterion = torch.nn.CrossEntropyLoss()

        # metric objects for calculating and averaging accuracy across batches
        self.train_acc = Accuracy(task="multiclass", num_classes=10)
        self.val_acc = Accuracy(task="multiclass", num_classes=10)
        self.test_acc = Accuracy(task="multiclass", num_classes=10)

        # for averaging loss across batches
        self.train_loss = MeanMetric()
        self.val_loss = MeanMetric()
        self.test_loss = MeanMetric()

        # for tracking best so far validation accuracy
        self.val_acc_best = MaxMetric()

    def forward(self, x: torch.Tensor):
        return self.net(x)

    def on_train_start(self):
        # by default lightning executes validation step sanity checks before training starts,
        # so we need to make sure val_acc_best doesn't store accuracy from these checks
        self.val_acc_best.reset()

    def model_step(self, batch: Any):
        x, y = batch
        logits = self.forward(x)
        loss = self.criterion(logits, y)
        preds = torch.argmax(logits, dim=1)
        return loss, preds, y

    def training_step(self, batch: Any, batch_idx: int):
        loss, preds, targets = self.model_step(batch)

        # update and log metrics
        self.train_loss(loss)
        self.train_acc(preds, targets)
        self.log("train/loss", self.train_loss, on_step=False, on_epoch=True, prog_bar=True)
        self.log("train/acc", self.train_acc, on_step=False, on_epoch=True, prog_bar=True)

        # we can return here dict with any tensors
        # and then read it in some callback or in `training_epoch_end()` below
        # remember to always return loss from `training_step()` or backpropagation will fail!
        return {"loss": loss, "preds": preds, "targets": targets}

    def training_epoch_end(self, outputs: List[Any]):
        # `outputs` is a list of dicts returned from `training_step()`

        # Warning: when overriding `training_epoch_end()`, lightning accumulates outputs from all batches of the epoch
        # this may not be an issue when training on mnist
        # but on larger datasets/models it's easy to run into out-of-memory errors

        # consider detaching tensors before returning them from `training_step()`
        # or using `on_train_epoch_end()` instead which doesn't accumulate outputs

        pass

    def validation_step(self, batch: Any, batch_idx: int):
        loss, preds, targets = self.model_step(batch)

        # update and log metrics
        self.val_loss(loss)
        self.val_acc(preds, targets)
        self.log("val/loss", self.val_loss, on_step=False, on_epoch=True, prog_bar=True)
        self.log("val/acc", self.val_acc, on_step=False, on_epoch=True, prog_bar=True)

        return {"loss": loss, "preds": preds, "targets": targets}

    def validation_epoch_end(self, outputs: List[Any]):
        acc = self.val_acc.compute()  # get current val acc
        self.val_acc_best(acc)  # update best so far val acc
        # log `val_acc_best` as a value through `.compute()` method, instead of as a metric object
        # otherwise metric would be reset by lightning after each epoch
        self.log("val/acc_best", self.val_acc_best.compute(), prog_bar=True)

    def test_step(self, batch: Any, batch_idx: int):
        loss, preds, targets = self.model_step(batch)

        # update and log metrics
        self.test_loss(loss)
        self.test_acc(preds, targets)
        self.log("test/loss", self.test_loss, on_step=False, on_epoch=True, prog_bar=True)
        self.log("test/acc", self.test_acc, on_step=False, on_epoch=True, prog_bar=True)

        return {"loss": loss, "preds": preds, "targets": targets}

    def test_epoch_end(self, outputs: List[Any]):
        pass

    def configure_optimizers(self):
        """Choose what optimizers and learning-rate schedulers to use in your optimization.
        Normally you'd need one. But in the case of GANs or similar you might have multiple.

        Examples:
            https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#configure-optimizers
        """
        optimizer = self.hparams.optimizer(params=self.parameters())
        if self.hparams.scheduler is not None:
            scheduler = self.hparams.scheduler(optimizer=optimizer)
            return {
                "optimizer": optimizer,
                "lr_scheduler": {
                    "scheduler": scheduler,
                    "monitor": "val/loss",
                    "interval": "epoch",
                    "frequency": 1,
                },
            }
        return {"optimizer": optimizer}
```

```python
class SimpleDenseNet(nn.Module):
    def __init__(
        self,
        input_size: int = 784,
        lin1_size: int = 256,
        lin2_size: int = 256,
        lin3_size: int = 256,
        output_size: int = 10,
    ):
        super().__init__()

        self.model = nn.Sequential(
            nn.Linear(input_size, lin1_size),
            nn.BatchNorm1d(lin1_size),
            nn.ReLU(),
            nn.Linear(lin1_size, lin2_size),
            nn.BatchNorm1d(lin2_size),
            nn.ReLU(),
            nn.Linear(lin2_size, lin3_size),
            nn.BatchNorm1d(lin3_size),
            nn.ReLU(),
            nn.Linear(lin3_size, output_size),
        )

    def forward(self, x):
        batch_size, channels, width, height = x.size()

        # (batch, 1, width, height) -> (batch, 1*width*height)
        x = x.view(batch_size, -1)

        return self.model(x)
```
### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):  Master .zip 1.9.0 pre-release candidate
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0): 2.0 nightly
#- Python version (e.g., 3.10): 3.10
#- OS (e.g., Linux): Linux
#- CUDA/cuDNN version: v8.7.0
#- GPU models and configuration: RTX 3090
#- How you installed Lightning(`conda`, `pip`, source): pip 
#- Running environment of LightningApp (e.g. local, cloud): local
```

</details>

### More info

Error
```
[2023-01-17 10:56:40,209][src.utils.utils][INFO] - Output dir: /home/johnny/Projects/lightning-hydra-template/logs/train/runs/2023-01-17_10-56-39
[2023-01-17 10:56:40,209][src.utils.utils][INFO] - Closing loggers...
Traceback (most recent call last):
  File "/home/johnny/Projects/lightning-hydra-template/src/train.py", line 127, in main
    metric_dict, _ = train(cfg)
  File "/home/johnny/Projects/lightning-hydra-template/src/utils/utils.py", line 48, in wrap
    raise ex
  File "/home/johnny/Projects/lightning-hydra-template/src/utils/utils.py", line 38, in wrap
    metric_dict, object_dict = task_func(cfg=cfg)
  File "/home/johnny/Projects/lightning-hydra-template/src/train.py", line 102, in train
    trainer.fit(model=model, datamodule=datamodule, ckpt_path=cfg.get("ckpt_path"))
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/trainer/trainer.py", line 608, in fit
    call._call_and_handle_interrupt(
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/trainer/call.py", line 38, in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/trainer/trainer.py", line 650, in _fit_impl
    self._run(model, ckpt_path=self.ckpt_path)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/trainer/trainer.py", line 1103, in _run
    results = self._run_stage()
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/trainer/trainer.py", line 1182, in _run_stage
    self._run_train()
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/trainer/trainer.py", line 1195, in _run_train
    self._run_sanity_check()
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/trainer/trainer.py", line 1267, in _run_sanity_check
    val_loop.run()
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/loops/loop.py", line 199, in run
    self.advance(*args, **kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/loops/dataloader/evaluation_loop.py", line 152, in advance
    dl_outputs = self.epoch_loop.run(self._data_fetcher, dl_max_batches, kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/loops/loop.py", line 199, in run
    self.advance(*args, **kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/loops/epoch/evaluation_epoch_loop.py", line 137, in advance
    output = self._evaluation_step(**kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/loops/epoch/evaluation_epoch_loop.py", line 234, in _evaluation_step
    output = self.trainer._call_strategy_hook(hook_name, *kwargs.values())
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/trainer/trainer.py", line 1485, in _call_strategy_hook
    output = fn(*args, **kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/lightning/pytorch/strategies/strategy.py", line 390, in validation_step
    return self.model.validation_step(*args, **kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 212, in _fn
    return fn(*args, **kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 333, in catch_errors
    return callback(frame, cache_size, hooks)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 480, in _convert_frame
    result = inner_convert(frame, cache_size, hooks)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 103, in _fn
    return fn(*args, **kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 94, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 339, in _convert_frame_assert
    return _compile(
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 400, in _compile
    out_code = transform_code_object(code, transform)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/bytecode_transformation.py", line 341, in transform_code_object
    transformations(instructions, code_options)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 387, in transform
    tracer.run()
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1692, in run
    super().run()
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 538, in run
    and self.step()
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 501, in step
    getattr(self, inst.opname)(inst)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 307, in wrapper
    return inner_fn(self, inst)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 974, in CALL_FUNCTION
    self.call_function(fn, args, {})
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 435, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 285, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 255, in call_function
    return super(UserFunctionVariable, self).call_function(tx, args, kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 88, in call_function
    return tx.inline_user_function_return(
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 471, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1770, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1826, in inline_call_
    tracer.run()
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 538, in run
    and self.step()
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 501, in step
    getattr(self, inst.opname)(inst)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1039, in LOAD_ATTR
    result = BuiltinVariable(getattr).call_function(
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py", line 322, in call_function
    result = handler(tx, *args, **kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py", line 699, in call_getattr
    return obj.var_getattr(tx, name).add_options(options)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/nn_module.py", line 131, in var_getattr
    return VariableBuilder(tx, NNModuleSource(source))(subobj)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/builder.py", line 169, in __call__
    return self._wrap(value).clone(**self.options())
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/builder.py", line 465, in _wrap
    return UserFunctionVariable(
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/base.py", line 28, in __call__
    obj.__post_init__(*args, **kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/base.py", line 281, in __post_init__
    VariableTracker.apply(
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/base.py", line 99, in apply
    result = fn(value.clone(**updated_dict))
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/base.py", line 65, in clone
    return self.__class__(**args)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/base.py", line 27, in __call__
    obj = type.__call__(cls, *args, **kwargs)
  File "/home/johnny/anaconda3/envs/lightning-hydra-template/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 110, in __init__
    assert isinstance(
AssertionError: expected FunctionType found method <bound method MNISTLitModule.forward of MNISTLitModule(
  (net): SimpleDenseNet(
    (model): Sequential(
      (0): Linear(in_features=784, out_features=64, bias=True)
      (1): BatchNorm1d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (2): ReLU()
      (3): Linear(in_features=64, out_features=128, bias=True)
      (4): BatchNorm1d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (5): ReLU()
      (6): Linear(in_features=128, out_features=64, bias=True)
      (7): BatchNorm1d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (8): ReLU()
      (9): Linear(in_features=64, out_features=10, bias=True)
    )
  )
  (criterion): CrossEntropyLoss()
  (train_acc): MulticlassAccuracy()
  (val_acc): MulticlassAccuracy()
  (test_acc): MulticlassAccuracy()
  (train_loss): MeanMetric()
  (val_loss): MeanMetric()
  (test_loss): MeanMetric()
  (val_acc_best): MaxMetric()
)>

from user code:
   File "/home/johnny/Projects/lightning-hydra-template/src/models/mnist_module.py", line 96, in validation_step
    loss, preds, targets = self.model_step(batch)
  File "/home/johnny/Projects/lightning-hydra-template/src/models/mnist_module.py", line 64, in model_step
    logits = self.forward(x)

Set torch._dynamo.config.verbose=True for more information


You can suppress this exception and fall back to eager by setting:
    torch._dynamo.config.suppress_errors = True

```

**Comment by andrey-klochkov-liftoff** [comment]
I'm seeing the same problem. I tried Lightning 1.9.0 and the master version.

**Comment by RaulSimpetru** [comment]
I also got this error today and just managed to solve it.
My understanding of the error is basically that `torch.compile` does not expect an NN to have a method that calls `self.forward` (expected FunctionType found method <bound method MNISTLitModule.forward of MNISTLitModule).

I solved the problem by "splitting" the actual architecture into a new class that inherits from `torch.nn.Module`.   
This is basically a vanilla Pytorch NN that can then be used in the Pytorch Lightning class as any other module.   
In init you can then call `torch.compile` on the vanilla NN and use it as a normal module.
 
 On the example provided I think (did not check) that this should solve it:
 ```python
 self.net = torch.compile(net)
 ```
 
 TL;DR: You can call `torch.compile` only on vanilla pytorch modules.

**Comment by johnnynunez** [comment]
> I also got this error today and just managed to solve it. My understanding of the error is basically that `torch.compile` does not expect an NN to have a method that calls `self.forward` (expected FunctionType found method <bound method MNISTLitModule.forward of MNISTLitModule).
> 
> I solved the problem by "splitting" the actual architecture into a new class that inherits from `torch.nn.Module`. This is basically a vanilla Pytorch NN that can then be used in the Pytorch Lightning class as any other module. In init you can then call `torch.compile` on the vanilla NN and use it as a normal module.
> 
> On the example provided I think (did not check) that this should solve it:
> 
> ```python
> self.net = torch.compile(net)
> ```
> 
> TL;DR: You can call `torch.compile` only on vanilla pytorch modules.

Do you have this error with this code?
https://github.com/ashleve/lightning-hydra-template

**Comment by carmocca** [comment]
This is fixed in master!

**Comment by st11279** [comment]
> This is fixed in master!
I'm seeing the same problem(Lightning 1.9.5) . Does this problem only be fixed in master or  some higher version be fixed at the same time? maybe 2.0.0 or 2.0.1. And which commit can fix this problem?

**Comment by carmocca** [comment]
I fixed it with this change: https://github.com/Lightning-AI/lightning/pull/16764/files#diff-464bb1669084536765b359fdb7734bea5ef6a1bb9e1e54c0a3069b8bc3718e4eR58

And it's available in 2.0.0 and 2.0.1

**Comment by carmocca** [comment]
This is because we have a hacky patch for it in PL 2.0.0 which was supposedly fixed in PyTorch 2.1.0. We avoid the patch in that case https://github.com/Lightning-AI/lightning/blob/ae8838b0a16efce7d4126b207870b60c54c67041/src/lightning/pytorch/utilities/compile.py#L60

Maybe there was a regression in PyTorch again.

Downgrading to PyTorch 2.0.0 should "fix" it for you @nicobonne

**Comment by nicobonne** [comment]
@carmocca thank you for pointing this out. 

I tried with PyTorch 2.0.0, and different PL versions (2.0.0, 2.0.1 and 2.0.1.post0). The error now is different but similar to the previous one:

```
AssertionError: expected FunctionType found _lru_cache_wrapper <functools._lru_cache_wrapper object at 0x7fd535c41430>

from user code:
   File "/home/nico/miniconda3/lib/python3.10/site-packages/lightning/fabric/utilities/distributed.py", line 91, in _distributed_available
    return torch.distributed.is_available() and torch.distributed.is_initialized() or _tpu_distributed()
  File "/home/nico/miniconda3/lib/python3.10/site-packages/lightning/fabric/accelerators/tpu.py", line 127, in _tpu_distributed
    if not TPUAccelerator.is_available():


Set torch._dynamo.config.verbose=True for more information


You can suppress this exception and fall back to eager by setting:
    torch._dynamo.config.suppress_errors = True

```

It seems another type mismatch from Fabric this time. 
Do you think this deserves a separate issue on PL?

**Comment by mrembalski** [comment]
I also encounter the same issue as @nicobonne. I'm using PyTorch `2.0.0+cu117`and Lightning `2.0.1`.

**Comment by mrembalski** [comment]
In my case I resolved the issue by adding `sync_dist=False` to `log_dict` in `LightningModule`. 

It seems that when `sync_dist=True` the strategy was checking `_tpu_distributed()` which called `TPUAccelerator.is_available()` and the `is_available()` method has the lru cache wrapper. 

By setting `sync_dist=False` (it is probably the default value) this won't be checked and the function won't be called. 

This is a temporary fix - it disables syncing in distributed environment (which I currently don't use, so it's not a big problem as of now).

**Comment by samils7** [comment]
I encounter the same problem with @mrembalski and I don't want to set `sync_dist=False` since I make my trainings in DDP. Any suggestion how to fix it?

**Comment by thesofakillers** [comment]
So does this mean we Lightning < 2 is not compatible with Torch >= 2? Any solution to users that depend on previous versions of lightning?
